tests/test_moe/test_top1.py

```diff
@@ -7,82 +7,81 @@
 from colossalai.context import ParallelMode
 from colossalai.core import global_context as gpc
 from colossalai.utils import free_port, get_current_device
-from colossalai.nn.layer.moe import Top1Router, MoeLayer
-from colossalai.global_variables import moe_env
+from colossalai.nn.layer.moe import Top1Router, MoeLayer, Experts
+from colossalai.core import MOE_CONTEXT
 
-BATCH_SIZE = 32
+BATCH_SIZE = 16
 NUM_EXPERTS = 4
-CONFIG = dict(parallel=dict(moe=dict(size=4)))
+CONFIG = dict()
 
 
-def check_equal(A, B, atol=1e-06):
-    assert torch.allclose(A, B, rtol=0, atol=atol) is True
+def check_equal(tensor_a, tensor_b, atol=1e-06):
+    assert torch.allclose(tensor_a, tensor_b, rtol=0, atol=atol) is True
 
 
 def run_routing(rank, world_size, port, rs=2, hidden_size=128, data_type=torch.float32):
-    colossalai.launch(config=CONFIG, rank=rank, world_size=world_size, host='localhost', port=port, backend='nccl')
+    torch.backends.cuda.matmul.allow_tf32 = False    # Here we do not need TF32, since it brings absolute error on results
 
-    # torch.set_printoptions(precision=30)
-    torch.backends.cuda.matmul.allow_tf32 = False
+    colossalai.launch(config=CONFIG, rank=rank, world_size=world_size, host='localhost', port=port, backend='nccl')
     local_rank = gpc.get_local_rank(ParallelMode.GLOBAL)
-    torch.manual_seed(rs + local_rank)
-    moe_env.reset_loss()
+
+    MOE_CONTEXT.setup(42)    # MOE environment initialization
+    MOE_CONTEXT.reset_loss()
+    torch.manual_seed(rs + local_rank)    # set each process has different random seed
+
+    # get randomized data
     tokens = torch.randn(BATCH_SIZE, hidden_size, dtype=data_type, device=get_current_device(), requires_grad=True)
-    # print(f"tokens:\n{tokens}")
-    router = Top1Router(1)
-    layer = MoeLayer(hidden_size, NUM_EXPERTS, router, nn.Identity())
+
+    expert = Experts(nn.Identity, NUM_EXPERTS)
+    layer = MoeLayer(hidden_size, NUM_EXPERTS, Top1Router(capacity_factor_train=1.0), expert)

```

Comment:
why not merge test_top1.py with test_top2.py?
The only diff is the parameters Top1Router.